text/0201-scope-attributes.md

```diff
@@ -0,0 +1,209 @@
+# Introduce Scope Attributes
+
+This OTEP adds attributes to the Scope of a telemetry emitter (e.g. Tracer, Meter, LogEmitter).
+
+## Motivation
+
+There are a few reasons why adding Scope attributes is a good idea:
+
+- There are 2 known use cases where Scope attributes can solve specific problems:
+  - Add support for [Meter "short_name"](https://github.com/open-telemetry/opentelemetry-specification/pull/2422),
+    represented as an attribute of Meter's Scope.
+  - Add support for differentiating the type of data emitted from the scopes that belong
+    to different data domains, e.g. profiling data emitted as log records or client-side
+    data emitted as log records needs to be differentiated so that it can be easily
+    routed and processed differently in the backends. We don't have a good way to handle
+    this today. The type of the data can be recorded as an attribute Logger's Scope.
+- It makes Scope consistent with the other primary data types: Resource, Span, Metric,
+  LogRecord.
+
+See additional [discussion here](https://github.com/open-telemetry/opentelemetry-specification/issues/2450).
+
+## Summary
+
+The following is the summary of proposed changes:
+
+- We will extend OpenTelemetry API to allow specifying Scope attributes when obtaining a
+  Tracer, Meter or LogEmitter. Scope attributes will be optional.
+- We will add `attributes` field to the [InstrumentationScope](https://github.com/open-telemetry/opentelemetry-proto/blob/88faab1197a2a105c7da659951e94bc951d37ab9/opentelemetry/proto/common/v1/common.proto#L83)
+  message of OTLP.
+- We will specify that Telemetry emitted via a Scope-ed Tracer, Meter or LogEmitter will
```

Comment:
If there are multiple Meter instances under the same MeterProvider, and these Meter instances have different Scope attributes, it could affect the aggregation logic. Also the user might want to configure if they want to have the Scope attributes be reported as metrics dimensions (e.g. when exporting to Prometheus) at all or not.

Probably need to do some prototype to have a meaningful conversation. I think it won't block this OTEP 👍

Reply:
Partial Go prototype for traces is here: https://github.com/open-telemetry/opentelemetry-go/pull/2884

I am resolving this comment since you said it won't block the OTEP. Feel free to reopen if needed.